### PR TITLE
Fixes 251228

### PR DESCRIPTION
**Pull request by belliottsmith**
InMemoryRangeIndex: Cassandra's secondary range index integration appears to be faulty in some way, producing sync points that have missing dependencies. Introduce a burn-test integrated in-memory implementation we can be more efficiently confident in.
Also:
     - SemiSyncValue
     - Simulate CFR loading/listening
     - KeyDeps.forEach
     - Timestamp.tryParse
Fix:
     - Minor burn test edge case when cancelling bootstrap and resuming
     - NotifyWaitingOn stillExecutes->stillWaitsOn
     - Privileged fast path commit failure should not send stable message
     - DefaultLocalListeners iterators do not handle multiple waiters on the same transaction correctly
Improve:
     - Don't compute Deps for SyncPoint Accept or Recover phase
     - Catchup should report bounds to progress log

### CODE REVIEW

File: accord-core/src/main/java/accord/impl/cfr/IdEntry.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import accord.primitives.Ranges;
+import accord.primitives.SaveStatus;
+import accord.primitives.Status;
+import accord.primitives.Timestamp;
+import accord.primitives.TxnId;
+import accord.utils.Invariants;
+
+public abstract class IdEntry extends TxnId
+{
+    static final int SAVE_SATUS_SHIFT = Status.Durability.ENCODING_BITS;
```
Comment (ifesdjeen) [inline accord-core/src/main/java/accord/impl/cfr/IdEntry.java]: nit: "satus"

File: accord-core/src/main/java/accord/impl/cfr/IdSingleEntry.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public IdEntry id()
+    {
+        return this;
+    }
+
+    @Override
+    IdEntry copy()
+    {
+        accord.impl.cfr.IdSingleEntry copy = new accord.impl.cfr.IdSingleEntry(this, range);
```
Comment (ifesdjeen) [inline accord-core/src/main/java/accord/impl/cfr/IdSingleEntry.java]: nit: do not need to fully qualify here

File: accord-core/src/main/java/accord/impl/cfr/InMemoryRangeSummaryIndex.java [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+            }
+            return f.apply(id, accumulator);
+        }
+    }
+
+    public void update(Command prev, Command updated, boolean force)
+    {
+        if (!force
+            && updated.saveStatus() == prev.saveStatus()
+            && (updated.saveStatus().known.executeAt().isDecided() && !prev.saveStatus().known.executeAt().isDecided())
```
Comment (ifesdjeen) [inline accord-core/src/main/java/accord/impl/cfr/InMemoryRangeSummaryIndex.java]: how can isDecided be different if save statuses are identity-equal?

**Review by ifesdjeen — APPROVED** [review]
